code/mobile/mobile-common/PVRSockets.cpp

```diff
@@ -355,19 +416,30 @@ void PVRStartReceiveStreams(uint16_t port) {
         catch (exception &e) {
             PVR_DB_I("[PVRStartReceiveStreams] caught Exception: " + to_string(e.what()));
         }
+        catch(...) {
+            PVR_DB("PVRSockets_PVRStartReceiveStreams:: Caught Exception: from Generic Handler");
+        }
     });
 }
 
 void PVRStopStreams() {
-    // talker sends disconnects at segue
-    delMtx.lock();
-    if (videoSvc)
-        videoSvc->stop(); //todo: use mutex
-    delMtx.unlock();
-
-    if (strThr) {
-        strThr->join();
-        delete strThr;
-        strThr = nullptr;
+    try {
+        // talker sends disconnects at segue
+        delMtx.lock();
+        if (videoSvc)
+            videoSvc->stop(); //todo: use mutex
+        delMtx.unlock();
+
+        if (strThr) {
+            strThr->join();
+            delete strThr;
+            strThr = nullptr;
+        }
+    }
+    catch(exception e) {
+        PVR_DB("PVRSockets_PVRStopStreams:: Caught Exception: " + string(e.what()));
+    }
+    catch(...) {
+        PVR_DB("PVRSockets_PVRStopStreams:: Caught Exception: from Generic Handler");
```

Comment:
Can we remove Generic catch block ? because even if it catches this type of exception, we will not be getting any solid information except that "Caught Exception: from Generic Handler", not helpful in debugging. Rather, we can let the App crash and let Android `debuggerd` collect stacktrace, and we can request client to upload tombstones, which will be more useful rather than "Caught Exception: from Generic Handler" without any message.